### PR TITLE
[MRG] API: Remove datasets.py

### PR DESCRIPTION
**Pull request by sappelhoff**
as discussed here https://github.com/mne-tools/mne-bids/issues/466#issuecomment-653868427

- `fetch_brainvision_testing_data` can just be removed
- `fetch_faces_data` needs to be replaced by some other NON-BIDS data ... for our "group conversion" example

As replacements for `fetch_faces_data` I suggest to use either:

- https://mne.tools/dev/generated/mne.datasets.limo.load_data.html#mne.datasets.limo.load_data

or

- https://mne.tools/dev/generated/mne.datasets.hf_sef.data_path.html#mne.datasets.hf_sef.data_path

they both have at least 2 subjects, so we can easily display group conversion functionality.

... or the eegbci data, which is probably nicest

Opinions?

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [x] All comments resolved
- [ ] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

**Comment by hoechenberger** [comment]
> As replacements for `fetch_faces_data` I suggest to use either:
> 
> * https://mne.tools/dev/generated/mne.datasets.limo.load_data.html#mne.datasets.limo.load_data
> 
> or
> 
> * https://mne.tools/dev/generated/mne.datasets.hf_sef.data_path.html#mne.datasets.hf_sef.data_path
> 
> they both have at least 2 subjects, so we can easily display group conversion functionality.
> 
> Opinions?

I'm fine with either – I've never worked with those datasets, but if they contain >= 2 participants, they should be good for our purposes!

**Comment by hoechenberger** [comment]
LGTM, feel free to merge once CIs go green

File: doc/whats_new.rst [hunk 1]
```diff
@@ -135,7 +136,7 @@ Changelog
 Bug
 ~~~
 
-- Fixed bug in :func:`mne_bids.datasets.fetch_faces_data` where downloading multiple subjects was impossible, by `Stefan Appelhoff`_ (`#262 <https://github.com/mne-tools/mne-bids/pull/262>`_)
+- Fixed bug in ``mne_bids.datasets.fetch_faces_data`` where downloading multiple subjects was impossible, by `Stefan Appelhoff`_ (`#262 <https://github.com/mne-tools/mne-bids/pull/262>`_)
```
Comment (jasmainak) [inline doc/whats_new.rst:139]: why the change to :func: ?

**Comment by jasmainak** [comment]
besides nitpicks, LGTM

File: doc/whats_new.rst (hunk 1 above)
Comment (sappelhoff) [reply doc/whats_new.rst:139]: the func needs to go, because it's no longer a func and thus the link would break. This is how Eric told me to do it in MNE-Python, so I just did the same here

**Comment by hoechenberger** [comment]
Thanks @sappelhoff!

File: doc/whats_new.rst (hunk 1 above)
Comment (jasmainak) [reply doc/whats_new.rst:139]: okay I see, thanks @sappelhoff
